src/digraph-cyclicity.jl

```diff
@@ -0,0 +1,439 @@
+abstract Visitor
+
+"""
+```transitiveclosure!(dg::DiGraph, selflooped = false)```
+
+Compute the transitive closure of a directed graph, using the Floyd-Warshall
+algorithm.
+
+Version of the function that modifies the original graph.
+
+Note: This is an O(V^3) algorithm. 
+
+# Arguments
+* `dg`: the directed graph on which the transitive closure is computed.
+* `selflooped`: whether self loop should be added to the directed graph,
+default to `false`.
+"""
+function transitiveclosure!(dg::DiGraph, selflooped = false)
+    if selflooped
+        for k in vertices(dg)
+            for i in vertices(dg)
```

Comment:
As far as I know, it does not work by iterating over vertices only. It is a Floyd-Warshall algorithm and it should, as said before, an optimization is to use Dijskra instead of FW.